src/tags/for_block.rs

```diff
@@ -118,13 +108,9 @@ fn parser_err<T>(expected: &str, actual: Option<&Token>) -> Result<T> {
   Err(Error::Parser(format!("Expected {}, found {:?}", expected, actual)))
 }
 
-/// Extracts an attribute with an integer value from the token stream 
+/// Extracts an attribute with an integer value from the token stream
 fn int_attr<'a>(args: &mut Iter<'a, Token>) -> Result<Option<usize>> {
-    match args.next() {
-        Some(&Colon) => (),
-        x => return parser_err(":", x)
-    };
-
+    try!(expect(args, Colon));
     match args.next() {
         Some(&NumberLiteral(ref n)) => Ok(Some(*n as usize)),
         x => return parser_err("number", x)
```

Comment:
Can't you replace `parser_err` with `Error::parser` now? (and delete the `parser_err fn?)